src/explore-education-statistics-frontend/src/services/permalinkService.ts

```diff
@@ -47,22 +50,163 @@ export interface Permalink {
   created: string;
   fullTable: FullTable;
   configuration: {
-    tableHeadersConfig?: TableHeadersFormValues;
+    tableHeadersConfig: TableHeadersFormValues;
   };
 }
 
+const reverseMapTableHeadersConfig = (
+  {
+    columns,
+    rows,
+    columnGroups,
+    rowGroups,
+  }: {
+    columns: FilterOption[];
+    columnGroups: FilterOption[][];
+    rows: FilterOption[];
+    rowGroups: FilterOption[][];
+  },
+  fullTableSubjectMeta: FullTableMeta,
+) => {
+  /**
+   * config filters only contain `values`.
+   * This function remaps the config from the filters in the subjectMeta
+   */
+  let mappedRows = [];
+  let mappedColumns = [];
+
+  const mappedRowGroups: (LocationFilter | CategoryFilter | any)[][] = [];
+  const mappedColumnGroups: (LocationFilter | CategoryFilter | any)[][] = [];
+
+  const initialValue = columns[0].value;
+  // rows/columns can only be TimePeriods / Indicators
+  // eslint-disable-next-line no-restricted-globals
+  if (isNaN(Number(initialValue))) {
+    // if NaN then timePeriod
+    mappedColumns = columns.map(({ value }) => {
+      return fullTableSubjectMeta.timePeriodRange.find(
+        timePeriod => `${timePeriod.year}_${timePeriod.code}` === value,
+      );
+    });
+    mappedRows = rows.map(({ value }) => {
+      return fullTableSubjectMeta.indicators.find(
+        indicator => indicator.value === value,
+      );
+    });
+  } else {
+    mappedRows = rows.map(({ value }) => {
+      return fullTableSubjectMeta.timePeriodRange.find(
+        timePeriod => `${timePeriod.year}_${timePeriod.code}` === value,
+      );
+    });
+    mappedColumns = columns.map(({ value }) => {
+      return fullTableSubjectMeta.indicators.find(
+        indicator => indicator.value === value,
+      );
+    });
+  }
+
+  // rowGroups/columnGroups can only be filters and locations
+  const locationAndFilterGroups: (LocationFilter | CategoryFilter)[][] = [
+    ...Object.values(
+      transformTableMetaFiltersToCategoryFilters(fullTableSubjectMeta.filters),
+    ),
+    fullTableSubjectMeta.locations,
+  ];
+
+  rowGroups.forEach((rowGroup, index) => {
+    let currentIndex = 0;
+    for (let i = 0; i < locationAndFilterGroups.length; i += 1) {
+      if (
+        locationAndFilterGroups[i].find(
+          element => element.value === rowGroup[0].value,
+        )
+      ) {
+        currentIndex = i;
+        break;
+      }
+    }
+
+    mappedRowGroups[index] = rowGroup.map(({ value }) =>
+      locationAndFilterGroups[currentIndex].find(
+        element => element.value === value,
+      ),
+    );
+    locationAndFilterGroups.splice(currentIndex, 1);
```

Comment:
Wouldn't bother mutating `locationAndFilterGroups`. I'm assuming you're worrying about performance, but the number of items that we have to iterate over should be pretty small.

Feels like a bit of a premature optimization and makes anyone reading it wonder why you've done it.